lib/recognition.gi

```diff
@@ -435,76 +512,105 @@ InstallGlobalFunction( ClassicalForms_QuadraticForm,
 
 #############################################################################
 ##
-#F  ClassicalForms_InvariantFormDual( <module>, <dmodule> )
+#F  ClassicalForms_InvariantFormDual( <gens_scalars> )
```

Comment:
Actually the `recog` package calls `ClassicalForms_InvariantFormDual`, so changing this here breaks `recog`. It would be great if we could avoid that. That could also include changing `recog` to call `ClassicalForms_InvariantFormDual` differently, but that's a lot more work and has to be carefully coordinated. Perhaps the old `ClassicalForms_InvariantFormDual` can be kept and the new code be added under a new name?

Or perhaps `recog` should be changed to not call these functions and instead call some alternative higher level functions, whatever they might be (perhaps they don't exist yet...)

Here is how `recog` calls this and a few other `forms` functions:
```gap

    # now try to find an invariant form
    if recognise.maybeDual then
        dmodule := ClassicalForms_GeneratorsWithoutScalarsDual(grp);
        if dmodule <> false then
            form := ClassicalForms_InvariantFormDual(module,dmodule);
            if form <> false  then
                Add( recognise.ClassicalForms,
                BilinearFormByMatrix(form[2], field));
                if Length(form)=4 then
                    recognise.QuadraticFormType := form[1];
                    recognise.QuadraticForm := QuadraticFormByMatrix(form[4]);
                fi;
                return false;
            else
                recognise.maybeDual := false;
            fi;
        fi;
    fi;

    if recognise.maybeFrobenius then
        fmodule := ClassicalForms_GeneratorsWithoutScalarsFrobenius(grp);
        if fmodule <> false then
            form := ClassicalForms_InvariantFormFrobenius(module,fmodule);
            if form <> false  then
                Add( recognise.ClassicalForms,
                     HermitianFormByMatrix(form[2], field));
               return false;
            else
                recognise.maybeFrobenius := false;
            fi;
        fi;
    fi;
```